tests/unit/challenges/test_aws_utils.py

```diff
@@ -0,0 +1,199 @@
+        self.challenge_host = ChallengeHost.objects.create(
+            user=self.user,
+            team_name=self.challenge_host_team,
+            status=ChallengeHost.ACCEPTED,
+            permissions=ChallengeHost.ADMIN,
+        )
+        self.challenge2 = Challenge.objects.create(
+            title="Test Challenge 2",
+            creator=self.challenge_host_team,
+            queue="test_queue_2",
+            start_date=timezone.now() - timedelta(days=2),
+            end_date=timezone.now() + timedelta(days=1),
+        )
+
+        self.client.force_authenticate(user=self.user)
+        self.ecs_client = boto3.client(
+            "ecs",
+            region_name=os.environ.get("AWS_DEFAULT_REGION", "us-east-1"),
+            aws_secret_access_key=os.environ.get("AWS_SECRET_ACCESS_KEY"),
+            aws_access_key_id=os.environ.get("AWS_ACCESS_KEY_ID"),
+        )
+
+    def set_challenge_workers(self, challenge, num_workers):
+        challenge.workers = num_workers
+        challenge.save()
+
+
+@mock_ecs
+class BaseAdminCallsClass(BaseTestClass):
+    def setUp(self):
+        super(BaseAdminCallsClass, self).setUp()
+
+        self.challenge3 = Challenge.objects.create(
+            title="Test Challenge 3",
+            creator=self.challenge_host_team,
+            queue="test_queue_3",
+            start_date=timezone.now() - timedelta(days=2),
+            end_date=timezone.now() + timedelta(days=1),
+        )
+        self.ecs_client.create_cluster(clusterName="cluster")
+        self.client_token = "abc123"
+
+    @classmethod
+    def queryset(cls, pklist):
+        queryset = Challenge.objects.filter(pk__in=pklist)
+        queryset = sorted(queryset, key=lambda i: pklist.index(i.pk))
+        return queryset
+
+
+class TestRestartWorkers(BaseAdminCallsClass):
+
+    def setUp(self):
+        super(TestRestartWorkers, self).setUp()
+
+        self.example_error = "Example Error Description"
+        self.response_OK = {
+            "ResponseMetadata": {"HTTPStatusCode": HTTPStatus.OK},
+        }
+        self.response_BAD_REQUEST = {
+            "ResponseMetadata": {"HTTPStatusCode": HTTPStatus.BAD_REQUEST},
+            "Error": self.example_error
+        }
+
+    def sm_side_effect(self, client, challenge, num_of_tasks, force_new_deployment):
+        self.set_challenge_workers(challenge, num_of_tasks)
+        return self.response_OK
+
+    @mock.patch("challenges.aws_utils.service_manager")
+    def test_restart_workers_when_all_challenges_have_active_workers(self, mock_sm):
+        mock_sm.side_effect = self.sm_side_effect
+
+        self.set_challenge_workers(self.challenge, 1)
+        self.set_challenge_workers(self.challenge2, 1)
+        self.set_challenge_workers(self.challenge3, 1)
+
+        pklist = [self.challenge.pk, self.challenge2.pk, self.challenge3.pk]
+        queryset = super(TestRestartWorkers, self).queryset(pklist)
+
+        expected_count = 3
+        expected_num_of_workers = [1, 1, 1]
```

Comment:
Can this `1` be declared as a class variable so that we can use the variable name where needed

Reply:
@Ram81 I think it is better if we keep it this way. As it states the number of workers and could be any number (0, 1, 2, 3, etc.) the number of variables would be too much to handle. 
What do you think?

Reply:
@nikochiko but the number is based on the number of challenges right? so can't we use that to compute the number of workers?

Reply:
Also as I can see in the code above you are setting the number of workers for every challenge. So I think we can just create 1 variable with a value that can be used in both setting the number of worker and using that in `expected_num_of_workers`. Let me know if this makes sense

Reply:
@Ram81 Okay, I think we can create a variable for the first case. (so something like `self.one_active_worker`?  Let me know if I have the right idea).
For the second case though, I want to clarify that it is the `expected_count` and not `expected_num_of_workers`. Both are different in that the expected count describes the number of workers successfully restarted during the process. So that can be different from the num_of_workers. For example in this case: https://github.com/Cloud-CV/EvalAI/pull/2604/files#diff-cc74cf6292b78dc74ea17124c792900eR192-R193

Reply:
> but the number is based on the number of challenges right? so can't we use that to compute the number of workers?

I'm sorry I didn't get what you meant by computing the number of workers. Can you please elaborate?

Reply:
If you are referring to computing the `expected_count`, I want to clarify that it is only the number of workers successfully restarted and not the number of active workers. It can be different for example in this case: https://github.com/Cloud-CV/EvalAI/pull/2604/files#diff-cc74cf6292b78dc74ea17124c792900eR192-R193